### PR TITLE
FieldBuilders now works top-level

### PR DESCRIPTION
**Pull request by thosakwe**
Until now, you'd get an error when trying to use a `FieldBuilder` at the top-level, because it extends `TopLevelMixin`. However, the `FieldBuilder` class is already capable of building top-level, so there is no need to throw the error.

The commit is small. :)

### CODE REVIEW

**Comment by natebosch** [comment]
Can we use a separate PR for the `asThrow` change? these are orthogonal and we can discuss them separately

**Comment by thosakwe** [comment]
@natebosch Sure. How do you propose I do that though? Is that possible with Git, or...?

**Comment by natebosch** [comment]
Yup, possible with Git, you will want to have a separate branch per pull request.

Looks like your latest commit should probably merge cleanly against `master` on it's own, so it shouldn't be too hard.

1. Make sure `dart-lang/code_builder` is a remote, say `upstream`
2. Start a new branch for the second pull request. `git co upstream/master -b as-throw`
3. Cherry-pick the asThrow commit (latest on your `master` branch) into this new branch (may have to resolve conflicts if any exists. `git cherry-pick master`
4. Open a pull request against your branch `as-throw` for this feature
5. Go back and set up this pull request so that it doesn't have these changes. `git co master` then `git revert HEAD` should do it.

**Comment by matanlurey** [comment]
+1, please split into two commits.

**Comment by matanlurey** [comment]
Any update here?

**Comment by thosakwe** [comment]
Will fix tonight

**Comment by matanlurey** [comment]
@thosakwe Ping.

**Comment by matanlurey** [comment]
@thosakwe Ping. I'll close if I don't receive an update by EOW.

**Comment by thosakwe** [comment]
Hey, sorry for the delay, got busy and sort of forgot about this PR. I'll definitely get this in.

**Comment by thosakwe** [comment]
@natebosch @matanlurey All right, I was able to remove the `asThrow` commits. I'll start an issue first before submitting a PR for the `asThrow` stuff. 👍

File: lib/src/tokens.dart [hunk 1]
```diff
@@ -187,6 +187,9 @@ final Token $static = new KeywordToken(Keyword.STATIC, 0);
 /// The `this` token.
 final Token $this = new KeywordToken(Keyword.THIS, 0);
 
+/// The `throw` token.
```
Comment (matanlurey) [inline lib/src/tokens.dart]: Just remove this from your PR and you're good to go.

Comment (thosakwe) [inline lib/src/tokens.dart]: Sorry, just saw this. Will do.

**Comment by matanlurey** [comment]
Closed as this is integrated in another PR now due to stalling out.

**Pull request closed without merging**